roc.go

```diff
@@ -0,0 +1,62 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import "sort"
+
+type byX struct {
+	x []float64
+	c []bool
+}
+
+func (a byX) Len() int { return len(a.x) }
+func (a byX) Swap(i, j int) {
+	a.x[i], a.x[j] = a.x[j], a.x[i]
+	a.c[i], a.c[j] = a.c[j], a.c[i]
+}
+func (a byX) Less(i, j int) bool { return a.x[i] < a.x[j] }
+
+// ROC returns paired false positive rate (FPR) and true positive
+// rate (TPR) values which describe the relative (or receiver)
+// operator chracteristic (ROC) curve obtained when y is
+// treated as a binary classifier for pos.
+// For a given cutoff value, observations corresponding to
+// entries in y greater than the cutoff are classified as
+// true, and these assigned class labels are compared to the
+// true class labels in pos to calculate FPR and TPR for all
+// possible cutoffs.
+// More details about ROC curves are available at
+// https://en.wikipedia.org/wiki/Receiver_operating_characteristic
+// Note that ROC will modify both inputs -- sorting them in order
+// of increasing y values.
+func ROC(y []float64, pos []bool) (tpr, fpr []float64) {
```

Comment:
Done. Also changed a bunch of other variable names to be more concise/ prettier.
Is it ok to use gotFPR instead of the perhaps more usual gotFpr?